docs/configuration.md

```diff
@@ -0,0 +1,297 @@
+# Cardano SL configuration
+
+This document describes Cardano SL configuration. Configuration is
+stored in YAML format, conventional name for the file is
+`configuration.yaml`. This file contains multiple configurations, each
+one is associated with a key. Almost all executables accept two
+options to specify configuration: `--configuration-file` is used to
+specify path to configuration and `--configuration-key` specifies key
+in this configuration. An example of configuration can be found in
+file `lib/configuration.yaml`.
+
+## Genesis
+
+Configuration is used to specify genesis data used by the node. It is
+part of `core` configuration (it's accessible by key
+`<configuration-key>.core.genesis`). There are two ways to specify
+genesis data.
+
+* One way is to provide genesis data itself, stored in json
+  format. Genesis data explicitly states balances of all addresses,
+  all bootstrap stakeholders, heavyweight delegation certificates,
+  etc. It makes it clearer which genesis is used, but it doesn't
+  describe _how_ this genesis was constructed. Also it requires more
+  steps to create genesis data, than to create a spec. Current design
+  implies that this mechanism should be used only for mainnet where we
+  want to have full understanding of which addresses have which
+  stake. The format is demonstrated below. `hash` is the hash (Blake2b
+  256 bits) of canonically encoded `mainnet-genesis.json` file. It can
+  be computed manually using `scripts/js/genesis-hash.js` script.
```

Comment:
I wonder if there's consensus to drop the JS code and switch to a single Haskell code/tool for this hash.

Reply:
cc @domenkozar -- based on the DevOps standup review.

Reply:
We can certainly have a Haskell tool if that makes things easier, it'd only be about 5 lines long too.

George does have a point that for transparency it is useful to keep an obviously independent implementation for others to use to verify our genesis block, but that doesn't mean we need to use it in our workflow if it is not convenient.